### ISSUE TITLE
Accessibility issues with va-date component

### ISSUE THREAD

**Issue by jaredcunha** [department-of-veterans-affairs/vets-design-system-documentation#1337]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

I noticed a couple potential accessibility issues with the date field:
1. Hint text does not seem to have a programatic link to the hint text
2. The select boxes have an `aria-label` of "Please enter two digits for the month/day".  With the current select box, this has a couple related issues for screen reader useres:
   1. Month uses month name, not numbers, so typing numbers does not work here.
   2. The text instructs user to enter two digits for day, but some days are a single digit.
   3. If the user does not select a month first, the listbox is empty. The user might benefit from some instructions to select a month first before selecting a day if they've skipped the month field. 
   4. This seems to have mixed results depending on how you navigate the page. Tab navigation, VoiceOver announces "Month, menu pop up collapsed, Please enter two digits for the month". In keyboard navigation controls, reads "Group, please enter two digits for the month" before digging further into the component to get to the dropdown.
<img width="856" alt="Screen Shot 2022-11-14 at 10 03 37 AM" src="https://user-images.githubusercontent.com/1094951/201693918-56088f71-8ed2-4aee-b939-84897a4e1393.png">

## What I expected to happen
- Hint text should be accessible via tab navigation, perhaps programmatically linked with `aria-describedby`
- Instructions should reflect what user actually needs to do.

## Reproducing
- Formation version: current
- Device: (e.g. iPhone 8, Macbook): Macbook
- Browser: (e.g. Firefox, IE 11): Chrome/VoiceOver

Steps to reproduce:

1. Can be tested with using screen reader on `va-date`. 

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [ ] Other

**Comment by davidakennedy** [comment]
I tried to add this issue to #1071 since it calls out some of the same things. Also, noted the things that @jaredcunha found that aren't called out in that older issue. Good finds, and all related! Hopefully, I've explained how they can be combined. Weigh in if I haven't. If I have, we could probably close this issue.

**Comment by jaredcunha** [comment]
@davidakennedy I think you've summarized the items in this ticket accurately in #1071 well.

**Comment by SarahKay8** [comment]
Closing - Summary of this bug has been addressed in ticket [#1071](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1071). Thanks DK! Will comment findings and enhancements there.
